server/desktop/src/main/java/dev/slimevr/desktop/platform/SteamVRBridge.kt

```diff
@@ -0,0 +1,423 @@
+package dev.slimevr.desktop.platform
+
+import dev.slimevr.VRServer
+import dev.slimevr.VRServer.Companion.getNextLocalTrackerId
+import dev.slimevr.VRServer.Companion.instance
+import dev.slimevr.bridge.BridgeThread
+import dev.slimevr.config.BridgeConfig
+import dev.slimevr.desktop.platform.ProtobufMessages.*
+import dev.slimevr.protocol.rpc.settings.RPCSettingsHandler
+import dev.slimevr.tracking.trackers.Tracker
+import dev.slimevr.tracking.trackers.TrackerPosition
+import dev.slimevr.tracking.trackers.TrackerPosition.Companion.getByTrackerRole
+import dev.slimevr.tracking.trackers.TrackerRole
+import dev.slimevr.tracking.trackers.TrackerRole.Companion.getById
+import dev.slimevr.tracking.trackers.TrackerUtils.getTrackerForSkeleton
+import dev.slimevr.util.ann.VRServerThread
+import io.eiren.util.collections.FastList
+import solarxr_protocol.rpc.StatusData
+import solarxr_protocol.rpc.StatusDataUnion
+import solarxr_protocol.rpc.StatusSteamVRDisconnectedT
+
+abstract class SteamVRBridge(
+	server: VRServer,
+	threadName: String?,
+	bridgeName: String?,
+	protected val bridgeSettingsKey: String,
+	protected val shareableTrackers: List<Tracker>,
+) : ProtobufBridge(bridgeName),
+	Runnable {
+	protected val runnerThread: Thread = Thread(this, threadName)
+	protected val config: BridgeConfig = server.configManager.vrConfig.getBridge(bridgeSettingsKey)
+
+	@VRServerThread
+	override fun startBridge() {
+		for (tr in shareableTrackers) {
+			val role = tr.trackerPosition?.trackerRole
+			changeShareSettings(
+				role,
+				config.getBridgeTrackerRole(role, false),
+			)
+		}
+		runnerThread.start()
+	}
+
+	@VRServerThread
+	override fun getShareSetting(role: TrackerRole): Boolean {
+		for (tr in shareableTrackers) {
+			if (tr.trackerPosition?.trackerRole == role) {
+				return sharedTrackers.contains(tr)
+			}
+		}
+		return false
+	}
+
+	override fun updateShareSettingsAutomatically(): Boolean {
+		if (!config.automaticSharedTrackersToggling) return false
+		// Enable waist if skeleton has an spine tracker
+		changeShareSettings(TrackerRole.WAIST, instance.humanPoseManager.skeleton.hasSpineTracker)
+		val skeleton = instance.humanPoseManager.skeleton
```

Comment:
Why not use the variable here lol
```suggestion
		val skeleton = instance.humanPoseManager.skeleton
		// Enable waist if skeleton has an spine tracker
		changeShareSettings(TrackerRole.WAIST, skeleton.hasSpineTracker)
```